artemis-server/src/main/java/org/apache/activemq/artemis/core/protocol/ProtocolHandler.java

```diff
@@ -147,11 +156,26 @@ public void channelInactive(ChannelHandlerContext ctx) throws Exception {
             timeoutFuture.cancel(true);
             timeoutFuture = null;
          }
+         if (proxyAttributesSet) {
+            ctx.channel().attr(PROXY_PROTOCOL_SOURCE_ADDRESS).remove();
+            ctx.channel().attr(PROXY_PROTOCOL_SOURCE_PORT).remove();
+            ctx.channel().attr(PROXY_PROTOCOL_DESTINATION_ADDRESS).remove();
+            ctx.channel().attr(PROXY_PROTOCOL_DESTINATION_PORT).remove();
+            ctx.channel().attr(PROXY_PROTOCOL_VERSION).remove();
+         }
       }
 
       @Override
       public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception {
-         if (msg instanceof FullHttpRequest httpRequest) {
+         if (msg instanceof HAProxyMessage haProxyMessage) {
+            ctx.channel().attr(PROXY_PROTOCOL_SOURCE_ADDRESS).set(haProxyMessage.sourceAddress());
+            ctx.channel().attr(PROXY_PROTOCOL_SOURCE_PORT).set(Integer.toString(haProxyMessage.sourcePort()));
+            ctx.channel().attr(PROXY_PROTOCOL_DESTINATION_ADDRESS).set(haProxyMessage.destinationAddress());
+            ctx.channel().attr(PROXY_PROTOCOL_DESTINATION_PORT).set(Integer.toString(haProxyMessage.destinationPort()));
+            ctx.channel().attr(PROXY_PROTOCOL_VERSION).set(haProxyMessage.protocolVersion().toString());
+            proxyAttributesSet = true;
+            skipProxyBytes = true;
+         } else if (msg instanceof FullHttpRequest httpRequest) {
```

Comment:
Should `skipProxyBytes` be false for `FullHttpRequest`?

Reply:
Yes.